homeassistant/components/hassio/backup.py

```diff
@@ -463,6 +467,37 @@ async def async_resume_restore_progress_after_restart(
         on_progress: Callable[[RestoreBackupEvent | IdleEvent], None],
     ) -> None:
         """Check restore status after core restart."""
+        if not (restore_job_id := os.environ.get(RESTORE_JOB_ID_ENV)):
+            _LOGGER.debug("No restore job ID found in environment")
+            return
+
+        _LOGGER.debug("Found restore job ID %s in environment", restore_job_id)
+
+        @callback
+        def on_job_progress(data: Mapping[str, Any]) -> None:
+            """Handle backup restore progress."""
+            if data.get("done") is not True:
+                on_progress(
+                    RestoreBackupEvent(
+                        reason="", stage=None, state=RestoreBackupState.IN_PROGRESS
+                    )
+                )
+                return
+
+            success = True
+            state = (
+                RestoreBackupState.COMPLETED if success else RestoreBackupState.FAILED
```

Comment:
`success` is always true.

Reply:
Oh, right, the problem is that we don't check the status of supervisor restore jobs in hassio.
I've removed the useless condition for now ; in a follow-up, we should add checks for the status when the job ends both when core restarts (the code added in this PR) and when core doesn't restart (the existing code).

Reply:
Ok.